website_event_membership_restriction/views/event_templates.xml

```diff
@@ -43,9 +45,11 @@
             <attribute name="class">d-none</attribute>
         </xpath>
         <xpath expr="//form//div[@t-else='']//button" position="after">
+            <t t-set="user" t-value="request.env.user" />
+            <t t-set="partner" t-value="user.partner_id" />
             <t
                 t-set="is_disabled_for_non_members"
-                t-value="event.registration_membership_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'] or request.env.user.partner_id.membership_state not in ['paid', 'free'])"
+                t-value="event.registration_membership_only and (not user or (partner.commercial_partner_id.membership_state not in ['paid', 'free'] and partner.membership_state not in ['paid', 'free']))"
```

Comment:
Don't we have it already from the initial t-set?

Reply:
True, removed the duplicated variable assignation.